### PR TITLE
Fasterq dump

### PR DESCRIPTION
**Pull request by Maarten-vd-Sande**
**What problem is the PR solving / What's new?**
Switched to fasterq-dump. Should be faster according to random benchmark. Makes sense because parallel-fastq-dump is just fastq-dump in parallel vs an implemenatataion that is made for parallel.

**Checklist**
- [x] I made a PR to develop (not master)
- [x] If applicable: I updated the docs
- [x] I updated the CHANGELOG
- [x] These changes are covered by the tests

### CODE REVIEW

**Comment by Maarten-vd-Sande** [comment]
parallel-fastq dump makes use of $TMPDIR, but I believe that fasterq-dump makes use of some weird config file of them. TODO: check if this works!

**Comment by Maarten-vd-Sande** [comment]
Not sure if it's faster, but at least we switched to the newer fastq-dump / actively maintained fastq-dump.

File: seq2science/rules/get_fastq.smk [hunk 1]
```diff
@@ -176,17 +183,24 @@ rule sra2fastq_PE:
...
-        parallel-fastq-dump -s {input}/* -O $tmpdir {config[split]} \
-        --threads {threads} --gzip >> {log} 2>&1
+        # acquire the lock
+        (
+            flock --timeout 30 200 || exit 1
+            sleep 2
+        ) 200>{layout_cachefile_lock}
+
+        # dump
+        fasterq-dump -s {input}/* -O $tmpdir --threads {threads} --split-3 >> {log} 2>&1
```
Comment (siebrenf) [inline seq2science/rules/get_fastq.smk:193]: out and tmp directories should be on the same disk, something like this?

```
# setup tmp dirs
out={config[sra_dir]}/out/{wildcards.sample}
mkdir -p $out; trap "rm -rf $out" EXIT
tmp={config[sra_dir]}/tmp/{wildcards.sample}
mkdir -p $tmp; trap "rm -rf $tmp" EXIT

...

fasterq-dump -s {input}/* -O $out -t $tmp --threads {threads} --split-3 >> {log} 2>&1

...

for f in $(ls -1q $out | grep -oP "^[^_]+" | uniq); do
```

Comment (siebrenf) [inline seq2science/rules/get_fastq.smk:193]: nitpick: --split-3 seems the default for `fasterq-dump`

File: seq2science/schemas/config/download.schema.yaml [hunk 2]
```diff
@@ -28,13 +28,3 @@ properties:
...
-
-  # fastq-dump parameters
-  split:
-    description: fastq-dump parameters when splitting reads (paired-end)
-    default: --split-files --skip-technical --dumpbase --readids --clip --read-filter pass --defline-seq '@$ac.$si.$sg/$ri' --defline-qual '+'
-    type: string
-  splot:
-    description: fastq-dump parameters when splotting reads (single-end)
-    default: --split-spot  --skip-technical --dumpbase --readids --clip --read-filter pass --defline-seq '@$ac.$si.$sg/$ri' --defline-qual '+'
-    type: string
```
Comment (siebrenf) [inline seq2science/schemas/config/download.schema.yaml:40]: question: so these are all done by default now? Their wiki only mentioned split-spot and skip-technical.

**Review by siebrenf — CHANGES_REQUESTED** [review]
For fasterq-dump, the tmp dir is the current directory by default. I think we want this to be on the same location as the out dir.

File: seq2science/schemas/config/download.schema.yaml (hunk 2 above)
Comment (Maarten-vd-Sande) [reply seq2science/schemas/config/download.schema.yaml:40]: Yes, fasterq-dump has "reasonable" defaults apparantly

File: seq2science/rules/get_fastq.smk (hunk 1 above)
Comment (Maarten-vd-Sande) [reply seq2science/rules/get_fastq.smk:193]: I know, I just thought this is more clear in what the difference is between those two

**Comment by siebrenf** [comment]
While browsing their wiki I also found [this ascp warning](https://github.com/ncbi/sra-tools/wiki/Avoid-using-ascp-directly-for-downloads). Should we still support ASCP?

File: seq2science/rules/get_fastq.smk (hunk 1 above)
Comment (Maarten-vd-Sande) [reply seq2science/rules/get_fastq.smk:193]: Yeah that would make sense

Comment (siebrenf) [reply seq2science/rules/get_fastq.smk:193]: 👍

File: seq2science/schemas/config/download.schema.yaml (hunk 2 above)
Comment (siebrenf) [reply seq2science/schemas/config/download.schema.yaml:40]: nice! just wanted to check :)

**Comment by Maarten-vd-Sande** [comment]
Oh that is interesting! I didnt know about the ascp thing!! I guess we switch to prefetch then, however we can keep ascp for the ENA since it is actually much faster there.

**Comment by Maarten-vd-Sande** [comment]
>download the requested file:
The file is downloaded using Aspera if available on your system, or HTTPS otherwise.

They can not be any more unclear?! I guess what they mean is that the alias ascp points to the path of ascp?

**Comment by Maarten-vd-Sande** [comment]
No ascp with prefetch. Whatever, we can still download the fastqs directly from ena, and assume that if its not on ena as fastq then its not as sra either there..
https://github.com/ncbi/sra-tools/issues/255

Documentation is baaaad

**Comment by Maarten-vd-Sande** [comment]
Prefetch makes is much, much more clean!

**Review by siebrenf — APPROVED** [review]
👍

**Comment by Maarten-vd-Sande** [comment]
I downloaded 2 terabytes without problems!

**Pull request merged**